### ISSUE TITLE
How to reproduce the result for WSJ?

### ISSUE THREAD

**Issue by rm-rf0** [iesl/diora#1]
Hi,

Thanks for the inspiring work and the well written code!

The instruction in README is very clear, but it's for the NLI dataset. I wonder how can I reproduce the result for WSJ in the paper?

It seems that in order to train on WSJ, I need to prepare WSJ in space-delimited format, e.g.
```
not surprisingly , he sometimes bites .
and so it was on gray friday .
```
But I am not sure how can I provide the corresponding parse trees for these sentences?

If possible, can you also provide the parameters to reproduce the WSJ result in the paper?

**Comment by mrdrozdov** [comment]
Our WSJ results were achieved with training on NLI. The training hyperparams are described here: https://github.com/iesl/diora#training

```
python -m torch.distributed.launch --nproc_per_node=4 diora/scripts/train.py \
    --arch treelstm \
    --batch_size 128 \
    --data_type nli \
    --elmo_cache_dir ~/data/elmo \
    --emb elmo \
    --hidden_dim 400 \
    --k_neg 100 \
    --log_every_batch 100 \
    --lr 2e-3 \
    --normalize unit \
    --reconstruct_mode margin \
    --save_after 1000 \
    --train_filter_length 20 \
    --train_path ~/data/allnli.jsonl \
    --cuda --multigpu
```

After training, you can use a saved model checkpoint to parse WSJ. This is also in the README, but if you have a more specific question, I can try to answer it: https://github.com/iesl/diora#parsing

Correct! You will need to prepare WSJ in space-delimited format. Unfortunately, the WSJ corpus is not open source.

**Comment by rm-rf0** [comment]
Can I train the model on WSJ instead of NLI?

To do so, I just need to prepare the WSJ train in NLI format. Is that correct?

**Comment by rm-rf0** [comment]
And I wonder if the provided WSJ is in space-delimited format, how does this program know what is the gold parse tree for the corresponding sentence?

**Comment by mrdrozdov** [comment]
> Can I train the model on WSJ instead of NLI?

You can do this definitely (this is what I typically do). You can also use other data_types as described in the README:
- `txt` - One space-delimited sentence per line.
- `txt_id` - One space-delimited sentence per line, and the first token is the example id.

> how does this program know what is the gold parse tree for the corresponding sentence?

We don't provide any evaluation code at this time. Only parsing code. If needed, this repo https://github.com/harvardnlp/urnng#evaluation describes how to perform evaluation using EVALB.

**Comment by rm-rf0** [comment]
Thanks a lot! Sorry for the unclarified questions!

**Comment by bobwan1995** [comment]
Hi, can you get the same result on the WSJ dataset as the paper reports (F1 max 56.76 )? When I directly use the checkpoint the author provided in this repo to evaluate on WSJ, I can only obtain a Corpus F1 around 43 (Here I use the full PTB test set with sentences length larger than 1 and punctuation removed). I don't know whether I need to finetune it with WSJ dataset? Or I miss some details? Thank you so much!

**Comment by mrdrozdov** [comment]
You should parse, and remove punctuation after.

If you are willing to wait a little bit, then I can add the output trees to the repo.

**Comment by bobwan1995** [comment]
Thank you so much for your kind reply!
> You should parse, and remove punctuation after.

Do you mean I should keep the punctuations in the inference stage, and then remove them with postprocessing?

I would be grateful if you could add the output trees to the repo. I'm looking forward to the results!
